File: typescript/my-first-package/schemas/myFirstModule/dynamic.ts
// Module: schemas.myFirstModule.dynamic
import * as myFirstPackageSchemasMyFirstModuleStructures from "./structures";
import * as typelayerCommonDynamic from "typelayer-common";


export function dynamicGetMyEnumStructure(dynamicObject: typelayerCommonDynamic.Decoder): myFirstPackageSchemasMyFirstModuleStructures.MyEnumStructure {
    let constructor : string;
    let myEnumIntegerBuilder : myFirstPackageSchemasMyFirstModuleStructures.MyEnumInteger;
    let myEnumIntegerFieldName : string;
    let myEnumIntegerFieldsLeft : number;
    let myEnumIntegerInit : myFirstPackageSchemasMyFirstModuleStructures.MyEnumInteger;
    let myEnumIntegerMyTextFieldValue : number;
    let myEnumIntegerNewFieldsLeft : number;
    let myEnumTextBuilder : myFirstPackageSchemasMyFirstModuleStructures.MyEnumText;
    let myEnumTextFieldName : string;
    let myEnumTextFieldsLeft : number;
    let myEnumTextInit : myFirstPackageSchemasMyFirstModuleStructures.MyEnumText;
    let myEnumTextMyTextFieldValue : string;
    let myEnumTextNewFieldsLeft : number;
    let numFields : number;
    [constructor, numFields] = dynamicObject.decoderPopStruct("my_enum_integer");
    switch (constructor) {
        case "my_enum_integer":
            myEnumIntegerInit = myFirstPackageSchemasMyFirstModuleStructures.defaultMyEnumInteger();
            [myEnumIntegerFieldsLeft, myEnumIntegerBuilder] = [numFields, myEnumIntegerInit];
            while ((-1 < myEnumIntegerFieldsLeft)) {
                if ((0 < myEnumIntegerFieldsLeft)) {
                    myEnumIntegerNewFieldsLeft = myEnumIntegerFieldsLeft;
                } else {
                    myEnumIntegerNewFieldsLeft = dynamicObject.decoderPopHasMoreStruct();
                }
                if ((0 < myEnumIntegerNewFieldsLeft)) {
                    myEnumIntegerFieldName = dynamicObject.decoderPopField();
                    switch (myEnumIntegerFieldName) {
                        case "*":
                            throw new typelayerCommonDynamic.DynamicGetException("Constructor must be first field", {path: dynamicObject.decoderPath()});
                            break;
                        case "my_text_field":
                            myEnumIntegerMyTextFieldValue = dynamicObject.decoderPopInteger();
                            myEnumIntegerBuilder.myTextField = myEnumIntegerMyTextFieldValue;
                            
                            [myEnumIntegerFieldsLeft, myEnumIntegerBuilder] = [(myEnumIntegerNewFieldsLeft - 1), myEnumIntegerBuilder];
                            break;
                        default:
                            throw new typelayerCommonDynamic.DynamicGetException("Unknown Field", {path: dynamicObject.decoderPath()});
                    }
                } else {
                    [myEnumIntegerFieldsLeft, myEnumIntegerBuilder] = [-1, myEnumIntegerBuilder];
                }
            }
            return myEnumIntegerBuilder;
            break;
        case "my_enum_text":
            myEnumTextInit = myFirstPackageSchemasMyFirstModuleStructures.defaultMyEnumText();
            [myEnumTextFieldsLeft, myEnumTextBuilder] = [numFields, myEnumTextInit];
            while ((-1 < myEnumTextFieldsLeft)) {
                if ((0 < myEnumTextFieldsLeft)) {
                    myEnumTextNewFieldsLeft = myEnumTextFieldsLeft;
                } else {
                    myEnumTextNewFieldsLeft = dynamicObject.decoderPopHasMoreStruct();
                }
                if ((0 < myEnumTextNewFieldsLeft)) {
                    myEnumTextFieldName = dynamicObject.decoderPopField();
                    switch (myEnumTextFieldName) {
                        case "*":
                            throw new typelayerCommonDynamic.DynamicGetException("Constructor must be first field", {path: dynamicObject.decoderPath()});
                            break;
                        case "my_text_field":
                            myEnumTextMyTextFieldValue = dynamicObject.decoderPopString();
                            myEnumTextBuilder.myTextField = myEnumTextMyTextFieldValue;
                            
                            [myEnumTextFieldsLeft, myEnumTextBuilder] = [(myEnumTextNewFieldsLeft - 1), myEnumTextBuilder];
                            break;
                        default:
                            throw new typelayerCommonDynamic.DynamicGetException("Unknown Field", {path: dynamicObject.decoderPath()});
                    }
                } else {
                    [myEnumTextFieldsLeft, myEnumTextBuilder] = [-1, myEnumTextBuilder];
                }
            }
            return myEnumTextBuilder;
            break;
        default:
            throw new typelayerCommonDynamic.DynamicGetException("Unknown Constructor", {path: dynamicObject.decoderPath()});
    }
}

export function dynamicGetObjMyEnumStructure(dynamicObject: typelayerCommonDynamic.Decoder): myFirstPackageSchemasMyFirstModuleStructures.MyEnumStructure {
    let retObj : myFirstPackageSchemasMyFirstModuleStructures.MyEnumStructure;
    retObj = dynamicGetMyEnumStructure(dynamicObject);
    return retObj;
}

export function dynamicPutMyEnumStructure(dynamicObject: typelayerCommonDynamic.Encoder, inputObject: myFirstPackageSchemasMyFirstModuleStructures.MyEnumStructure): void {
    let myEnumStructureFieldMyEnumInteger : myFirstPackageSchemasMyFirstModuleStructures.MyEnumInteger;
    let myEnumStructureFieldMyEnumText : myFirstPackageSchemasMyFirstModuleStructures.MyEnumText;
    if (inputObject instanceof myFirstPackageSchemasMyFirstModuleStructures.MyEnumInteger) {
        let myEnumStructureFieldMyEnumInteger = inputObject;
        dynamicObject.encoderPutStruct("my_enum_integer", 1);
        dynamicObject.encoderPutField("my_text_field");
        dynamicObject.encoderPutInteger(myEnumStructureFieldMyEnumInteger.myTextField);
        dynamicObject.encoderPutFieldEnd(true);
        return;
    } else if (inputObject instanceof myFirstPackageSchemasMyFirstModuleStructures.MyEnumText) {
        let myEnumStructureFieldMyEnumText = inputObject;
        dynamicObject.encoderPutStruct("my_enum_text", 1);
        dynamicObject.encoderPutField("my_text_field");
        dynamicObject.encoderPutString(myEnumStructureFieldMyEnumText.myTextField);
        dynamicObject.encoderPutFieldEnd(true);
        return;
    } else {
        throw new Error(`Invalid type ${inputObject}`);
    }
}

export function dynamicGetMyFirstStructure(dynamicObject: typelayerCommonDynamic.Decoder): myFirstPackageSchemasMyFirstModuleStructures.MyFirstStructure {
    let constructor : string;
    let myFirstStructureBuilder : myFirstPackageSchemasMyFirstModuleStructures.MyFirstStructure;
    let myFirstStructureFieldName : string;
    let myFirstStructureFieldsLeft : number;
    let myFirstStructureInit : myFirstPackageSchemasMyFirstModuleStructures.MyFirstStructure;
    let myFirstStructureMyBooleanFieldValue : boolean;
    let myFirstStructureMyDoubleFieldValue : number;
    let myFirstStructureMyIntegerFieldValue : number;
    let myFirstStructureMyListFieldElementsLeft : number;
    let myFirstStructureMyListFieldInit : number;
    let myFirstStructureMyListFieldListAcc : Array<string>;
    let myFirstStructureMyListFieldListValue : string;
    let myFirstStructureMyListFieldNewFieldsLeft : number;
    let myFirstStructureMyListFieldValue : Array<string>;
    let myFirstStructureMyMapFieldElementsLeft : number;
    let myFirstStructureMyMapFieldInit : number;
    let myFirstStructureMyMapFieldKey : string;
    let myFirstStructureMyMapFieldMapAcc : Map<string, string>;
    let myFirstStructureMyMapFieldMapValue : string;
    let myFirstStructureMyMapFieldNewFieldsLeft : number;
    let myFirstStructureMyMapFieldValue : Map<string, string>;
    let myFirstStructureMyNestedFieldValue : myFirstPackageSchemasMyFirstModuleStructures.MyEnumStructure;
    let myFirstStructureMyTextFieldValue : string;
    let myFirstStructureNewFieldsLeft : number;
    let numFields : number;
    [constructor, numFields] = dynamicObject.decoderPopStruct("my_first_structure");
    switch (constructor) {
        case "my_first_structure":
            myFirstStructureInit = myFirstPackageSchemasMyFirstModuleStructures.defaultMyFirstStructure();
            [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [numFields, myFirstStructureInit];
            while ((-1 < myFirstStructureFieldsLeft)) {
                if ((0 < myFirstStructureFieldsLeft)) {
                    myFirstStructureNewFieldsLeft = myFirstStructureFieldsLeft;
                } else {
                    myFirstStructureNewFieldsLeft = dynamicObject.decoderPopHasMoreStruct();
                }
                if ((0 < myFirstStructureNewFieldsLeft)) {
                    myFirstStructureFieldName = dynamicObject.decoderPopField();
                    switch (myFirstStructureFieldName) {
                        case "*":
                            throw new typelayerCommonDynamic.DynamicGetException("Constructor must be first field", {path: dynamicObject.decoderPath()});
                            break;
                        case "my_boolean_field":
                            myFirstStructureMyBooleanFieldValue = dynamicObject.decoderPopBool();
                            myFirstStructureBuilder.myBooleanField = myFirstStructureMyBooleanFieldValue;
                            
                            [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [(myFirstStructureNewFieldsLeft - 1), myFirstStructureBuilder];
                            break;
                        case "my_double_field":
                            myFirstStructureMyDoubleFieldValue = dynamicObject.decoderPopDouble();
                            myFirstStructureBuilder.myDoubleField = myFirstStructureMyDoubleFieldValue;
                            
                            [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [(myFirstStructureNewFieldsLeft - 1), myFirstStructureBuilder];
                            break;
                        case "my_integer_field":
                            myFirstStructureMyIntegerFieldValue = dynamicObject.decoderPopInteger();
                            myFirstStructureBuilder.myIntegerField = myFirstStructureMyIntegerFieldValue;
                            
                            [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [(myFirstStructureNewFieldsLeft - 1), myFirstStructureBuilder];
                            break;
                        case "my_list_field":
                            myFirstStructureMyListFieldInit = dynamicObject.decoderPopList();
                            [myFirstStructureMyListFieldElementsLeft, myFirstStructureMyListFieldListAcc] = [myFirstStructureMyListFieldInit, []];
                            while ((-1 < myFirstStructureMyListFieldElementsLeft)) {
                                if ((0 < myFirstStructureMyListFieldElementsLeft)) {
                                    myFirstStructureMyListFieldNewFieldsLeft = myFirstStructureMyListFieldElementsLeft;
                                } else {
                                    myFirstStructureMyListFieldNewFieldsLeft = dynamicObject.decoderPopHasMoreList();
                                }
                                if ((0 < myFirstStructureMyListFieldNewFieldsLeft)) {
                                    myFirstStructureMyListFieldListValue = dynamicObject.decoderPopString();
                                    myFirstStructureMyListFieldListAcc.push(myFirstStructureMyListFieldListValue);
                                    
                                    [myFirstStructureMyListFieldElementsLeft, myFirstStructureMyListFieldListAcc] = [(myFirstStructureMyListFieldNewFieldsLeft - 1), myFirstStructureMyListFieldListAcc];
                                } else {
                                    [myFirstStructureMyListFieldElementsLeft, myFirstStructureMyListFieldListAcc] = [-1, myFirstStructureMyListFieldListAcc];
                                }
                            }
                            myFirstStructureMyListFieldValue = myFirstStructureMyListFieldListAcc;
                            myFirstStructureBuilder.myListField = myFirstStructureMyListFieldValue;
                            
                            [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [(myFirstStructureNewFieldsLeft - 1), myFirstStructureBuilder];
                            break;
                        case "my_map_field":
                            myFirstStructureMyMapFieldInit = dynamicObject.decoderPopMap();
                            [myFirstStructureMyMapFieldElementsLeft, myFirstStructureMyMapFieldMapAcc] = [myFirstStructureMyMapFieldInit, new Map()];
                            while ((-1 < myFirstStructureMyMapFieldElementsLeft)) {
                                if ((0 < myFirstStructureMyMapFieldElementsLeft)) {
                                    myFirstStructureMyMapFieldNewFieldsLeft = myFirstStructureMyMapFieldElementsLeft;
                                } else {
                                    myFirstStructureMyMapFieldNewFieldsLeft = dynamicObject.decoderPopHasMoreMap();
                                }
                                if ((0 < myFirstStructureMyMapFieldNewFieldsLeft)) {
                                    myFirstStructureMyMapFieldKey = dynamicObject.decoderPopKey();
                                    myFirstStructureMyMapFieldMapValue = dynamicObject.decoderPopString();
                                    myFirstStructureMyMapFieldMapAcc.set(myFirstStructureMyMapFieldKey, myFirstStructureMyMapFieldMapValue);
                                    
                                    [myFirstStructureMyMapFieldElementsLeft, myFirstStructureMyMapFieldMapAcc] = [(myFirstStructureMyMapFieldNewFieldsLeft - 1), myFirstStructureMyMapFieldMapAcc];
                                } else {
                                    [myFirstStructureMyMapFieldElementsLeft, myFirstStructureMyMapFieldMapAcc] = [-1, myFirstStructureMyMapFieldMapAcc];
                                }
                            }
                            myFirstStructureMyMapFieldValue = myFirstStructureMyMapFieldMapAcc;
                            myFirstStructureBuilder.myMapField = myFirstStructureMyMapFieldValue;
                            
                            [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [(myFirstStructureNewFieldsLeft - 1), myFirstStructureBuilder];
                            break;
                        case "my_nested_field":
                            myFirstStructureMyNestedFieldValue = dynamicGetMyEnumStructure(dynamicObject);
                            myFirstStructureBuilder.myNestedField = myFirstStructureMyNestedFieldValue;
                            
                            [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [(myFirstStructureNewFieldsLeft - 1), myFirstStructureBuilder];
                            break;
                        case "my_text_field":
                            myFirstStructureMyTextFieldValue = dynamicObject.decoderPopString();
                            myFirstStructureBuilder.myTextField = myFirstStructureMyTextFieldValue;
                            
                            [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [(myFirstStructureNewFieldsLeft - 1), myFirstStructureBuilder];
                            break;
                        default:
                            throw new typelayerCommonDynamic.DynamicGetException("Unknown Field", {path: dynamicObject.decoderPath()});
                    }
                } else {
                    [myFirstStructureFieldsLeft, myFirstStructureBuilder] = [-1, myFirstStructureBuilder];
                }
            }
            return myFirstStructureBuilder;
            break;
        default:
            throw new typelayerCommonDynamic.DynamicGetException("Unknown Constructor", {path: dynamicObject.decoderPath()});
    }
}

export function dynamicGetObjMyFirstStructure(dynamicObject: typelayerCommonDynamic.Decoder): myFirstPackageSchemasMyFirstModuleStructures.MyFirstStructure {
    let retObj : myFirstPackageSchemasMyFirstModuleStructures.MyFirstStructure;
    retObj = dynamicGetMyFirstStructure(dynamicObject);
    return retObj;
}

export function dynamicPutMyFirstStructure(dynamicObject: typelayerCommonDynamic.Encoder, inputObject: myFirstPackageSchemasMyFirstModuleStructures.MyFirstStructure): void {
    let myFirstStructureFieldMyFirstStructure : myFirstPackageSchemasMyFirstModuleStructures.MyFirstStructure;
    if (inputObject instanceof myFirstPackageSchemasMyFirstModuleStructures.MyFirstStructure) {
        let myFirstStructureFieldMyFirstStructure = inputObject;
        dynamicObject.encoderPutStruct("my_first_structure", 7);
        dynamicObject.encoderPutField("my_boolean_field");
        dynamicObject.encoderPutBool(myFirstStructureFieldMyFirstStructure.myBooleanField);
        dynamicObject.encoderPutFieldEnd(false);
        dynamicObject.encoderPutField("my_double_field");
        dynamicObject.encoderPutDouble(myFirstStructureFieldMyFirstStructure.myDoubleField);
        dynamicObject.encoderPutFieldEnd(false);
        dynamicObject.encoderPutField("my_integer_field");
        dynamicObject.encoderPutInteger(myFirstStructureFieldMyFirstStructure.myIntegerField);
        dynamicObject.encoderPutFieldEnd(false);
        dynamicObject.encoderPutField("my_list_field");
        dynamicObject.encoderPutList(myFirstStructureFieldMyFirstStructure.myListField.length);
        if ((0 === myFirstStructureFieldMyFirstStructure.myListField.length)) {
            dynamicObject.encoderPutElementEnd(true);
        } else {
            let myListFieldIndex : number = 0;
            for (let myListFieldElem of myFirstStructureFieldMyFirstStructure.myListField) {
                dynamicObject.encoderPutString(myListFieldElem);
                dynamicObject.encoderPutElementEnd((myListFieldIndex === (myFirstStructureFieldMyFirstStructure.myListField.length - 1)));
                
                myListFieldIndex++;
            }
            
        }
        dynamicObject.encoderPutFieldEnd(false);
        dynamicObject.encoderPutField("my_map_field");
        dynamicObject.encoderPutMap(myFirstStructureFieldMyFirstStructure.myMapField.size);
        if ((0 === myFirstStructureFieldMyFirstStructure.myMapField.size)) {
            dynamicObject.encoderPutValueEnd(true);
        } else {
            let myMapFieldValuesIndex : number = 0;
            for (let [myMapFieldValuesKey, myMapFieldValuesElem] of myFirstStructureFieldMyFirstStructure.myMapField) {
                dynamicObject.encoderPutKey(myMapFieldValuesKey);
                dynamicObject.encoderPutString(myMapFieldValuesElem);
                dynamicObject.encoderPutValueEnd((myMapFieldValuesIndex === (myFirstStructureFieldMyFirstStructure.myMapField.size - 1)));
                
                myMapFieldValuesIndex++;
            }
            
        }
        dynamicObject.encoderPutFieldEnd(false);
        dynamicObject.encoderPutField("my_nested_field");
        dynamicPutMyEnumStructure(dynamicObject, myFirstStructureFieldMyFirstStructure.myNestedField);
        dynamicObject.encoderPutFieldEnd(false);
        dynamicObject.encoderPutField("my_text_field");
        dynamicObject.encoderPutString(myFirstStructureFieldMyFirstStructure.myTextField);
        dynamicObject.encoderPutFieldEnd(true);
        return;
    } else {
        throw new Error(`Invalid type ${inputObject}`);
    }
}

File: python/my_first_package/my_first_package/schemas/my_first_module/dynamic.py
from . import structures as my_first_package_schemas_my_first_module_structures
from typelayer_common import dynamic as typelayer_common_dynamic
from dataclasses import dataclass
from typing import List, Dict, Optional


def dynamic_get_my_enum_structure(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_schemas_my_first_module_structures.MyEnumStructure:
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    (constructor, num_fields) = dynamic_object.decoder_pop_struct("my_enum_integer")
    if (constructor == "my_enum_integer"):
        my_enum_integer_init = my_first_package_schemas_my_first_module_structures.default_my_enum_integer()
        (my_enum_integer_fields_left, my_enum_integer_builder) = (num_fields, my_enum_integer_init)
        while ((-1 < my_enum_integer_fields_left)):
            if ((0 < my_enum_integer_fields_left)):
                my_enum_integer_new_fields_left = my_enum_integer_fields_left
            else:
                my_enum_integer_new_fields_left = dynamic_object.decoder_pop_has_more_struct()
            if ((0 < my_enum_integer_new_fields_left)):
                my_enum_integer_field_name = dynamic_object.decoder_pop_field()
                if (my_enum_integer_field_name == "*"):
                    raise typelayer_common_dynamic.DynamicGetException("Constructor must be first field", {"path": dynamic_object.decoder_path()});
                elif (my_enum_integer_field_name == "my_text_field"):
                    my_enum_integer_my_text_field_value = dynamic_object.decoder_pop_integer()
                    my_enum_integer_builder.my_text_field = my_enum_integer_my_text_field_value;
                    
                    (my_enum_integer_fields_left, my_enum_integer_builder) = ((my_enum_integer_new_fields_left - 1), my_enum_integer_builder)
                else:
                        raise typelayer_common_dynamic.DynamicGetException("Unknown Field", {"path": dynamic_object.decoder_path()});
            else:
                (my_enum_integer_fields_left, my_enum_integer_builder) = (-1, my_enum_integer_builder)
        return my_enum_integer_builder
    elif (constructor == "my_enum_text"):
        my_enum_text_init = my_first_package_schemas_my_first_module_structures.default_my_enum_text()
        (my_enum_text_fields_left, my_enum_text_builder) = (num_fields, my_enum_text_init)
        while ((-1 < my_enum_text_fields_left)):
            if ((0 < my_enum_text_fields_left)):
                my_enum_text_new_fields_left = my_enum_text_fields_left
            else:
                my_enum_text_new_fields_left = dynamic_object.decoder_pop_has_more_struct()
            if ((0 < my_enum_text_new_fields_left)):
                my_enum_text_field_name = dynamic_object.decoder_pop_field()
                if (my_enum_text_field_name == "*"):
                    raise typelayer_common_dynamic.DynamicGetException("Constructor must be first field", {"path": dynamic_object.decoder_path()});
                elif (my_enum_text_field_name == "my_text_field"):
                    my_enum_text_my_text_field_value = dynamic_object.decoder_pop_string()
                    my_enum_text_builder.my_text_field = my_enum_text_my_text_field_value;
                    
                    (my_enum_text_fields_left, my_enum_text_builder) = ((my_enum_text_new_fields_left - 1), my_enum_text_builder)
                else:
                        raise typelayer_common_dynamic.DynamicGetException("Unknown Field", {"path": dynamic_object.decoder_path()});
            else:
                (my_enum_text_fields_left, my_enum_text_builder) = (-1, my_enum_text_builder)
        return my_enum_text_builder
    else:
            raise typelayer_common_dynamic.DynamicGetException("Unknown Constructor", {"path": dynamic_object.decoder_path()});

def dynamic_get_obj_my_enum_structure(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_schemas_my_first_module_structures.MyEnumStructure:

    ret_obj = dynamic_get_my_enum_structure(dynamic_object)
    return ret_obj

def dynamic_put_my_enum_structure(dynamic_object: typelayer_common_dynamic.Encoder, input_object: my_first_package_schemas_my_first_module_structures.MyEnumStructure) -> None:
    
    
    if isinstance(input_object, my_first_package_schemas_my_first_module_structures.MyEnumInteger):
        my_enum_structure_field_my_enum_integer = input_object
        dynamic_object.encoder_put_struct("my_enum_integer", 1)
        dynamic_object.encoder_put_field("my_text_field")
        dynamic_object.encoder_put_integer(my_enum_structure_field_my_enum_integer.my_text_field)
        dynamic_object.encoder_put_field_end(True)
        return;
    elif isinstance(input_object, my_first_package_schemas_my_first_module_structures.MyEnumText):
        my_enum_structure_field_my_enum_text = input_object
        dynamic_object.encoder_put_struct("my_enum_text", 1)
        dynamic_object.encoder_put_field("my_text_field")
        dynamic_object.encoder_put_string(my_enum_structure_field_my_enum_text.my_text_field)
        dynamic_object.encoder_put_field_end(True)
        return;
    else:
        raise Exception(f"Invalid type {input_object}")

def dynamic_get_my_first_structure(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_schemas_my_first_module_structures.MyFirstStructure:
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    (constructor, num_fields) = dynamic_object.decoder_pop_struct("my_first_structure")
    if (constructor == "my_first_structure"):
        my_first_structure_init = my_first_package_schemas_my_first_module_structures.default_my_first_structure()
        (my_first_structure_fields_left, my_first_structure_builder) = (num_fields, my_first_structure_init)
        while ((-1 < my_first_structure_fields_left)):
            if ((0 < my_first_structure_fields_left)):
                my_first_structure_new_fields_left = my_first_structure_fields_left
            else:
                my_first_structure_new_fields_left = dynamic_object.decoder_pop_has_more_struct()
            if ((0 < my_first_structure_new_fields_left)):
                my_first_structure_field_name = dynamic_object.decoder_pop_field()
                if (my_first_structure_field_name == "*"):
                    raise typelayer_common_dynamic.DynamicGetException("Constructor must be first field", {"path": dynamic_object.decoder_path()});
                elif (my_first_structure_field_name == "my_boolean_field"):
                    my_first_structure_my_boolean_field_value = dynamic_object.decoder_pop_bool()
                    my_first_structure_builder.my_boolean_field = my_first_structure_my_boolean_field_value;
                    
                    (my_first_structure_fields_left, my_first_structure_builder) = ((my_first_structure_new_fields_left - 1), my_first_structure_builder)
                elif (my_first_structure_field_name == "my_double_field"):
                    my_first_structure_my_double_field_value = dynamic_object.decoder_pop_double()
                    my_first_structure_builder.my_double_field = my_first_structure_my_double_field_value;
                    
                    (my_first_structure_fields_left, my_first_structure_builder) = ((my_first_structure_new_fields_left - 1), my_first_structure_builder)
                elif (my_first_structure_field_name == "my_integer_field"):
                    my_first_structure_my_integer_field_value = dynamic_object.decoder_pop_integer()
                    my_first_structure_builder.my_integer_field = my_first_structure_my_integer_field_value;
                    
                    (my_first_structure_fields_left, my_first_structure_builder) = ((my_first_structure_new_fields_left - 1), my_first_structure_builder)
                elif (my_first_structure_field_name == "my_list_field"):
                    my_first_structure_my_list_field_init = dynamic_object.decoder_pop_list()
                    (my_first_structure_my_list_field_elements_left, my_first_structure_my_list_field_list_acc) = (my_first_structure_my_list_field_init, [])
                    while ((-1 < my_first_structure_my_list_field_elements_left)):
                        if ((0 < my_first_structure_my_list_field_elements_left)):
                            my_first_structure_my_list_field_new_fields_left = my_first_structure_my_list_field_elements_left
                        else:
                            my_first_structure_my_list_field_new_fields_left = dynamic_object.decoder_pop_has_more_list()
                        if ((0 < my_first_structure_my_list_field_new_fields_left)):
                            my_first_structure_my_list_field_list_value = dynamic_object.decoder_pop_string()
                            my_first_structure_my_list_field_list_acc.append(my_first_structure_my_list_field_list_value);
                            
                            (my_first_structure_my_list_field_elements_left, my_first_structure_my_list_field_list_acc) = ((my_first_structure_my_list_field_new_fields_left - 1), my_first_structure_my_list_field_list_acc)
                        else:
                            (my_first_structure_my_list_field_elements_left, my_first_structure_my_list_field_list_acc) = (-1, my_first_structure_my_list_field_list_acc)
                    my_first_structure_my_list_field_value = my_first_structure_my_list_field_list_acc
                    my_first_structure_builder.my_list_field = my_first_structure_my_list_field_value;
                    
                    (my_first_structure_fields_left, my_first_structure_builder) = ((my_first_structure_new_fields_left - 1), my_first_structure_builder)
                elif (my_first_structure_field_name == "my_map_field"):
                    my_first_structure_my_map_field_init = dynamic_object.decoder_pop_map()
                    (my_first_structure_my_map_field_elements_left, my_first_structure_my_map_field_map_acc) = (my_first_structure_my_map_field_init, {})
                    while ((-1 < my_first_structure_my_map_field_elements_left)):
                        if ((0 < my_first_structure_my_map_field_elements_left)):
                            my_first_structure_my_map_field_new_fields_left = my_first_structure_my_map_field_elements_left
                        else:
                            my_first_structure_my_map_field_new_fields_left = dynamic_object.decoder_pop_has_more_map()
                        if ((0 < my_first_structure_my_map_field_new_fields_left)):
                            my_first_structure_my_map_field_key = dynamic_object.decoder_pop_key()
                            my_first_structure_my_map_field_map_value = dynamic_object.decoder_pop_string()
                            my_first_structure_my_map_field_map_acc[my_first_structure_my_map_field_key] = my_first_structure_my_map_field_map_value
                            
                            (my_first_structure_my_map_field_elements_left, my_first_structure_my_map_field_map_acc) = ((my_first_structure_my_map_field_new_fields_left - 1), my_first_structure_my_map_field_map_acc)
                        else:
                            (my_first_structure_my_map_field_elements_left, my_first_structure_my_map_field_map_acc) = (-1, my_first_structure_my_map_field_map_acc)
                    my_first_structure_my_map_field_value = my_first_structure_my_map_field_map_acc
                    my_first_structure_builder.my_map_field = my_first_structure_my_map_field_value;
                    
                    (my_first_structure_fields_left, my_first_structure_builder) = ((my_first_structure_new_fields_left - 1), my_first_structure_builder)
                elif (my_first_structure_field_name == "my_nested_field"):
                    my_first_structure_my_nested_field_value = dynamic_get_my_enum_structure(dynamic_object)
                    my_first_structure_builder.my_nested_field = my_first_structure_my_nested_field_value;
                    
                    (my_first_structure_fields_left, my_first_structure_builder) = ((my_first_structure_new_fields_left - 1), my_first_structure_builder)
                elif (my_first_structure_field_name == "my_text_field"):
                    my_first_structure_my_text_field_value = dynamic_object.decoder_pop_string()
                    my_first_structure_builder.my_text_field = my_first_structure_my_text_field_value;
                    
                    (my_first_structure_fields_left, my_first_structure_builder) = ((my_first_structure_new_fields_left - 1), my_first_structure_builder)
                else:
                        raise typelayer_common_dynamic.DynamicGetException("Unknown Field", {"path": dynamic_object.decoder_path()});
            else:
                (my_first_structure_fields_left, my_first_structure_builder) = (-1, my_first_structure_builder)
        return my_first_structure_builder
    else:
            raise typelayer_common_dynamic.DynamicGetException("Unknown Constructor", {"path": dynamic_object.decoder_path()});

def dynamic_get_obj_my_first_structure(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_schemas_my_first_module_structures.MyFirstStructure:

    ret_obj = dynamic_get_my_first_structure(dynamic_object)
    return ret_obj

def dynamic_put_my_first_structure(dynamic_object: typelayer_common_dynamic.Encoder, input_object: my_first_package_schemas_my_first_module_structures.MyFirstStructure) -> None:

    if isinstance(input_object, my_first_package_schemas_my_first_module_structures.MyFirstStructure):
        my_first_structure_field_my_first_structure = input_object
        dynamic_object.encoder_put_struct("my_first_structure", 7)
        dynamic_object.encoder_put_field("my_boolean_field")
        dynamic_object.encoder_put_bool(my_first_structure_field_my_first_structure.my_boolean_field)
        dynamic_object.encoder_put_field_end(False)
        dynamic_object.encoder_put_field("my_double_field")
        dynamic_object.encoder_put_double(my_first_structure_field_my_first_structure.my_double_field)
        dynamic_object.encoder_put_field_end(False)
        dynamic_object.encoder_put_field("my_integer_field")
        dynamic_object.encoder_put_integer(my_first_structure_field_my_first_structure.my_integer_field)
        dynamic_object.encoder_put_field_end(False)
        dynamic_object.encoder_put_field("my_list_field")
        dynamic_object.encoder_put_list(len(my_first_structure_field_my_first_structure.my_list_field))
        if ((0 == len(my_first_structure_field_my_first_structure.my_list_field))):
            dynamic_object.encoder_put_element_end(True)
        else:
            my_list_field_index = 0;
            for my_list_field_elem in my_first_structure_field_my_first_structure.my_list_field:
                dynamic_object.encoder_put_string(my_list_field_elem)
                dynamic_object.encoder_put_element_end((my_list_field_index == (len(my_first_structure_field_my_first_structure.my_list_field) - 1)))
                
                my_list_field_index += 1
            
        dynamic_object.encoder_put_field_end(False)
        dynamic_object.encoder_put_field("my_map_field")
        dynamic_object.encoder_put_map(len(my_first_structure_field_my_first_structure.my_map_field))
        if ((0 == len(my_first_structure_field_my_first_structure.my_map_field))):
            dynamic_object.encoder_put_value_end(True)
        else:
            my_map_field_values_index = 0;
            for my_map_field_values_key, my_map_field_values_elem in my_first_structure_field_my_first_structure.my_map_field.items():
                dynamic_object.encoder_put_key(my_map_field_values_key)
                dynamic_object.encoder_put_string(my_map_field_values_elem)
                dynamic_object.encoder_put_value_end((my_map_field_values_index == (len(my_first_structure_field_my_first_structure.my_map_field) - 1)))
                
                my_map_field_values_index += 1
            
        dynamic_object.encoder_put_field_end(False)
        dynamic_object.encoder_put_field("my_nested_field")
        dynamic_put_my_enum_structure(dynamic_object, my_first_structure_field_my_first_structure.my_nested_field)
        dynamic_object.encoder_put_field_end(False)
        dynamic_object.encoder_put_field("my_text_field")
        dynamic_object.encoder_put_string(my_first_structure_field_my_first_structure.my_text_field)
        dynamic_object.encoder_put_field_end(True)
        return;
    else:
        raise Exception(f"Invalid type {input_object}")